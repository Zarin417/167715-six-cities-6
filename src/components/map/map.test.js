import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import Map from "./map";

const mockStore = configureStore({});

describe(`Test 'Map'`, () => {
  it(`Map should render correctly`, () => {
    const places = [{
      title: `Nice, cozy, warm big bed apartment`,
      price: 129,
      previewImage: ``,
      type: `room`,
      isFavorite: false,
      isPremium: false,
      rating: 3,
      id: 11,
      bedrooms: 1,
      description: ``,
      maxAdults: 1,
      location: {latitude: 52.36354, longitude: 4.911976, zoom: 16},
      images: [``],
      host: {id: 25, name: `Angelina`, isPro: false, avatarUrl: `img/avatar-angelina.jpg`},
      goods: [`Washer`],
      city: {name: `Amsterdam`, location: {latitude: 52.37454, longitude: 4.897976, zoom: 13}},
    }];

    const store = mockStore({
      CARD: {activeCard: 3}
    });

    const {container} = render(
        <Provider store={store}>
          <Map
            places={places}
            city={`Amsterdam`}
            placeInfo={places[0]}
          />
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
