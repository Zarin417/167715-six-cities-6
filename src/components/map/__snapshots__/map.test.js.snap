// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'Map' Map should render correctly 1`] = `
<div>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://c.basemaps.cartocdn.com/rastertiles/voyager/13/4207/2692.png"
              style="width: 256px; height: 256px; left: -117px; top: -10px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 81px; top: 105px; z-index: 105;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 81px; top: 105px; z-index: 105;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
          
          contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
